python/demo/poisson/demo_poisson.py

```diff
@@ -1,204 +1,236 @@
-# triangles, we do as follows ::
-
-# Create mesh and define function space
-mesh = create_rectangle(MPI.COMM_WORLD, ((0.0, 0.0), (2.0, 1.0)), (32, 16), CellType.triangle)
-V = FunctionSpace(mesh, ("Lagrange", 1))
-
-# The second argument to :py:class:`FunctionSpace
-# <dolfinx.fem.FunctionSpace>` is the finite element family, while the
-# third argument specifies the polynomial degree. Thus, in this case,
-# our space ``V`` consists of first-order, continuous Lagrange finite
-# element functions (or in order words, continuous piecewise linear
-# polynomials).
+# function space $V$ relative to this mesh. We create a rectangular mesh
+# using built-in function provided by class
+# {py:class}`create_rectangle<dolfinx.mesh.create_rectangle>`. In order
+# to create a mesh consisting of 32 x 16 squares with each square
+# divided into two triangles, we do as follows
+
+# +
+# Create mesh
+rect_mesh = mesh.create_rectangle(
+    comm=MPI.COMM_WORLD,
+    points=((0.0, 0.0), (2.0, 1.0)),
+    n=(32, 16),
+    cell_type=mesh.CellType.triangle,
+)
+
+# Define function space
+V = fem.FunctionSpace(rect_mesh, ("Lagrange", 1))
+# -
+
+# The second argument to {py:class}`FunctionSpace
+# <dolfinx.fem.FunctionSpace>` is a tuple consisting of `(family,
+# degree)`, where `family` is the finite element family, and `degree`
+# specifies the polynomial degree. Thus, in this case, our space `V`
+# consists of first-order, continuous Lagrange finite element functions
+# (or in order words, continuous piecewise linear polynomials).
 #
 # Next, we want to consider the Dirichlet boundary condition. A simple
 # Python function, returning a boolean, can be used to define the
-# boundary for the Dirichlet boundary condition (:math:`\Gamma_D`). The
-# function should return ``True`` for those points inside the boundary
-# and ``False`` for the points outside. In our case, we want to say that
-# the points :math:`(x, y)` such that :math:`x = 0` or :math:`x = 1` are
-# inside on the inside of :math:`\Gamma_D`. (Note that because of
-# rounding-off errors, it is often wise to instead specify :math:`x <
-# \epsilon` or :math:`x > 1 - \epsilon` where :math:`\epsilon` is a
-# small number (such as machine precision).)
+# boundary for the Dirichlet boundary condition ($\Gamma_D$). The
+# function should return `True` for those points inside the boundary and
+# `False` for the points outside. In our case, we want to say that the
+# points $(x, y)$ such that $x = 0$ or $x = 1$ are inside on the inside
+# of $\Gamma_D$. (Note that because of rounding-off errors, it is often
+# wise to instead specify $x < \epsilon$ or $x > 1 - \epsilon$ where
+# $\epsilon$ is a small number (such as machine precision).)
 #
 # Now, the Dirichlet boundary condition can be created using the class
-# :py:class:`DirichletBC <dolfinx.fem.bcs.DirichletBC>`. A
-# :py:class:`DirichletBC <dolfinx.fem.bcs.DirichletBC>` takes three
+# {py:class}`DirichletBC <dolfinx.fem.DirichletBC>`. A
```

Comment:
This documentation is outdated, as we now use the function `dolfinx.fem.dirichletbc` to create a DirichletBC of class
`dolfinx.fem.bcs.DirichletBCMetaClass`